code/game/objects/items/weapons/implants/implant_traitor.dm

```diff
@@ -45,11 +47,11 @@
 
 	// Create a new mindslave datum for the target with the user as their master.
 	mindslave_target.mind.add_antag_datum(new /datum/antagonist/mindslave(user.mind))
+	mindslave_UID = mindslave_target.mind.UID()
 	log_admin("[key_name_admin(user)] has mind-slaved [key_name_admin(mindslave_target)].")
 	return TRUE
 
 /obj/item/implant/traitor/removed(mob/target)
-	if(..())
-		target.mind.remove_antag_datum(/datum/antagonist/mindslave)
-		return TRUE
-	return FALSE
+	. = ..()
+	var/datum/mind/M = locateUID(mindslave_UID)
+	M.remove_antag_datum(/datum/antagonist/mindslave)
```

Comment:
there's probably some cleanup that should be done with `removed`'s returns in general, but I suppose this makes sense given that if `removed` gets called, the implant should, in fact, have been removed.